core-common/src/main/java/org/mapstruct/ObjectFactory.java

```diff
@@ -0,0 +1,46 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * By default beans are created during the mapping process with the default constructor.
+ * This method allows to mark a method as a factory method to create beans.
+ * The type of the factory method is determined by its return type. A factory is used
+ * to create a bean if their types match. If multiple factories match,
+ * the closest one is chosen.
+ * <p>
+ * The factory method can optionally retrieve the mapping sources by specifying
+ * them as parameters, exactly like mapping methods do. This allows the factory
+ * method to create target beans based on source beans, which opens up a variety
+ * of possibilities. For example, a factory method for entities can look up an
+ * EntityManager to check whether the entity already exists and then return its
+ * managed instance.
+ *
+ * @author Remo Meier
```

Comment:
Could you also add actual code examples here? Over the time we want to convert our javadoc to be more example driven, and I think we already have some in the existing annotations.